File: C++/Model/spuria.cpp
#include "spuria.h"

spuria::spuria(const polinomio& a, const polinomio& b):
    secondoGrado(a,b){
    polinomio temp = unifica();
    temp.ordina();
    temp.eliminaZeri();
    if(temp.getMonomio_pos(0).getGrado()!=razionale(1,1))
        throw erroreGrado();
    if(temp.getMonomio_pos(1).getGrado()!=razionale(2,1))
        throw erroreGrado();
}

string spuria::risolvi() const{
    polinomio temp = unifica();
    temp.ordina();
    temp.eliminaZeri();
    razionale r = razionale(-1,1)*temp.getMonomio_pos(0).getCoefficiente()/temp.getMonomio_pos(1).getCoefficiente();
    return "x= 0  &&  x= "+r.toString();
}

tipoEquazione spuria::getTipo() const{
    return tipoEquazione(3);
}

razionale spuria::getDelta() const{
    razionale a,b;
    polinomio temp = unifica();
    temp.ordina();
    temp.eliminaZeri();

    a=temp.getMonomio_pos(1).getCoefficiente();
    b=temp.getMonomio_pos(0).getCoefficiente();
    return b*b - razionale(4)*a;
}

File: Java/spuria.java
public class spuria extends secondoGrado{
    
    public spuria(polinomio sx, polinomio dx) throws erroreGrado{
        super(sx,dx); 
        polinomio temp = unifica();
        temp.ordina();
        if(temp.getMonomio_pos(0).getEsponente().diverso(new razionale(1)))
            throw new erroreGrado();
        if(temp.getMonomio_pos(1).getEsponente().diverso(new razionale(2)))
            throw new erroreGrado();
    }

    public String getRisultato(){
        polinomio temp = unifica();
        temp.ordina();
        razionale r = new razionale();
        razionale menoUno = new razionale(-1);
        try{
            r = razionale.moltiplicazioneRazionale(menoUno,razionale.divisioneRazionale(temp.getMonomio_pos(0).getCoefficiente(),temp.getMonomio_pos(1).getCoefficiente()));
        }
        catch(Exception e){}
        return "x= 0  &&  x= "+r.toString();
    }

    public String getTipo(){
        return "Equazione spuria di secondo grado";
    }

    public razionale getDelta(){
        razionale a = new razionale();
        razionale b = new razionale();
        razionale d = new razionale(4);
        polinomio temp = unifica();
        temp.ordina();
        a=temp.getMonomio_pos(1).getCoefficiente();
        b=temp.getMonomio_pos(0).getCoefficiente();
        try{
            return razionale.sottrRazionale(razionale.moltiplicazioneRazionale(b,b),razionale.moltiplicazioneRazionale(razionale.moltiplicazioneRazionale(d,a),new razionale(1)));
        }
        catch(Exception e){}
        return a;
    }
}